src/structures/BaseCommandInteraction.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const Interaction = require('./Interaction');
+const InteractionWebhook = require('./InteractionWebhook');
+const InteractionResponses = require('./interfaces/InteractionResponses');
+const { ApplicationCommandOptionTypes } = require('../util/Constants');
+
+/**
+ * Represents a command interaction.
+ * @extends {Interaction}
+ * @implements {InteractionResponses}
+ * @abstract
+ */
+class BaseCommandInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The channel this interaction was sent in
+     * @type {?(TextChannel|NewsChannel|DMChannel)}
```

Comment:
```suggestion
     * @type {?TextBasedChannels}
```
https://github.com/discordjs/discord.js/blob/47d2ef3e40165d846b5a901709b26dfffea46ef2/src/util/Constants.js#L464-L471